src/solvers/dgsem_tree/dg_2d_parabolic.jl

```diff
@@ -0,0 +1,652 @@
+          interfaces.u[2, v, i, interface] = u[v, i,          1, right_element]
+        end
+      end
+    end
+  end
+
+  # Calculate interface fluxes
+  @trixi_timeit timer() "interface flux" begin
+    @unpack surface_flux_values = cache_parabolic.elements
+    @unpack neighbor_ids, orientations = cache_parabolic.interfaces
+
+    @threaded for interface in eachinterface(dg, cache_parabolic)
+      # Get neighboring elements
+      left_id  = neighbor_ids[1, interface]
+      right_id = neighbor_ids[2, interface]
+
+      # Determine interface direction with respect to elements:
+      # orientation = 1: left -> 2, right -> 1
+      # orientation = 2: left -> 4, right -> 3
+      left_direction  = 2 * orientations[interface]
+      right_direction = 2 * orientations[interface] - 1
+
+      for i in eachnode(dg)
+        # Call pointwise Riemann solver
+        u_ll, u_rr = get_surface_node_vars(cache_parabolic.interfaces.u,
+                                           equations_parabolic, dg, i, interface)
+        flux = 0.5 * (u_ll + u_rr)
+
+        # Copy flux to left and right element storage
+        for v in eachvariable(equations_parabolic)
+          surface_flux_values[v, i, left_direction,  left_id]  = flux[v]
+          surface_flux_values[v, i, right_direction, right_id] = flux[v]
+        end
+      end
+    end
+  end
+
+  # Prolong solution to boundaries
+  @trixi_timeit timer() "prolong2boundaries" begin
+    @unpack boundaries = cache_parabolic
+    @unpack orientations, neighbor_sides = boundaries
+
+    @threaded for boundary in eachboundary(dg, cache_parabolic)
+      element = boundaries.neighbor_ids[boundary]
+
+      if orientations[boundary] == 1
+        # boundary in x-direction
+        if neighbor_sides[boundary] == 1
+          # element in -x direction of boundary
+          for l in eachnode(dg), v in eachvariable(equations_parabolic)
+            boundaries.u[1, v, l, boundary] = u[v, nnodes(dg), l, element]
+          end
+        else # Element in +x direction of boundary
+          for l in eachnode(dg), v in eachvariable(equations_parabolic)
+            boundaries.u[2, v, l, boundary] = u[v, 1,          l, element]
+          end
+        end
+      else # if orientations[boundary] == 2
+        # boundary in y-direction
+        if neighbor_sides[boundary] == 1
+          # element in -y direction of boundary
+          for l in eachnode(dg), v in eachvariable(equations_parabolic)
+            boundaries.u[1, v, l, boundary] = u[v, l, nnodes(dg), element]
+          end
+        else
+          # element in +y direction of boundary
+          for l in eachnode(dg), v in eachvariable(equations_parabolic)
+            boundaries.u[2, v, l, boundary] = u[v, l, 1,          element]
+          end
+        end
+      end
+    end
+  end
+
+  # Calculate boundary fluxes
+  @trixi_timeit timer() "boundary flux" begin
+    calc_gradient_boundary_flux!(cache_parabolic, t, boundary_conditions_parabolic,
```

Comment:
Here (and above), I'd again rename this in the `<general>_<specific>` pattern, i.e., something like `calc_boundary_flux_gradients!` and `calc_boundary_flux_divergence!`

Reply:
Done in 0fc9828f184b0f9b147af123ad7c5c676f2e1268